### PR TITLE
memory-pair-game

### PR DESCRIPTION
**Pull request by Iakow**
### memory-pair-game

[Demo](https://iakow.github.io/memory-pair-game/)
[Code base](https://github.com/Iakow/frontend-2021-homeworks/tree/pair-game/submissions/iakow/memory-pair-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by OleksiyRudenko** [comment]
@Iakow we do not need images for review. Submission of those makes cloning and sync-up time longer for other students (and for yourself).
image

Please remove images from this PR

**Comment by Iakow** [comment]
> @Iakow we do not need images for review. Submission of those makes cloning and sync-up time longer for other students (and for yourself).
> image
> 
> Please remove images from this PR

I'm sorry, done

**Comment by OleksiyRudenko** [comment]
@Iakow A really nice gameplay and UX 👍

**Comment by Iakow** [comment]
:warning: :warning: :warning: 
Refrain from reviewing for a few days, please, I will launch a totally different version.

**Comment by Iakow** [comment]
@babayK0 :hand:

File: submissions/iakow/memory-pair-game/js/game.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  }
+}
+
+function completeMatch() {
+  return new Promise((resolve) => {
+    const awaitDiscardingLastCards = 1400;
+
+    setTimeout(() => {
+      const time = new Date(Date.now() - matchStartTime);
+      const score = Math.round(100000000 / (matchMoves * time));
```
Comment (babayK0) [inline submissions/iakow/memory-pair-game/js/game.js:62]: 100000000 - magic number

File: submissions/iakow/memory-pair-game/js/game.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+function closeMismatchedPair() {
+  matchMoves++;
+  Render.animateClose(...openedCardsIDs).then(clearOpenedCards);
+}
+
+function discardMatchedPair() {
+  matchMoves++;
+  gameCards[openedCardsIDs[0]] = gameCards[openedCardsIDs[1]] = null;
```
Comment (babayK0) [inline submissions/iakow/memory-pair-game/js/game.js:94]: use destructuring (with index it is not obvious)
`const [firstId, secondId] = openedCardsIDs`

File: submissions/iakow/memory-pair-game/js/game.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  Render.animateDiscard(...openedCardsIDs).then(clearOpenedCards);
+}
+
+function checkIfLastPair() {
+  return gameCards.filter((card) => card != null).length === 2;
+}
```
Comment (babayK0) [inline submissions/iakow/memory-pair-game/js/game.js:100]:
```suggestion
function checkIfLastPair() {
  return gameCards.filter((card) => card).length === 2;
}
```

**Review by babayK0 — CHANGES_REQUESTED** [review]
Liked code structure and UI :+1: 
Watch my comments below

**Review by babayK0 — APPROVED** [review]
:+1:

**Comment by Iakow** [comment]
Thanks for review!

**Pull request merged**